### PR TITLE
update deps, remove shallow compare

### PR DESCRIPTION
**Pull request by chirag04**
cc @brentvatne @janicduplessis

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,28 +22,27 @@
...
-    "invariant": "^2.2.1",
-    "lodash": "^4.6.1",
+    "core-decorators": "^0.17.0",
+    "fbemitter": "^2.1.1",
+    "fbjs": "^0.8.12",
+    "hoist-non-react-statics": "^1.2.0",
+    "invariant": "^2.2.2",
+    "lodash": "^4.17.4",
     "mirror-creator": "^1.1.0",
-    "react-addons-shallow-compare": "15.3.2",
```
Comment (chirag04) [inline package.json]: it was a peer dep of tab-view. Which is not the case anymore. cc @satya164

File: package.json [hunk 2]
```diff
@@ -22,28 +22,27 @@
...
     "react-clone-referenced-element": "^1.0.1",
     "react-mixin": "^3.0.5",
-    "react-native-drawer-layout-polyfill": "^1.1.0",
-    "react-native-tab-view": "0.0.52",
-    "react-redux": "^4.4.5",
-    "react-static-container": "^1.0.0",
+    "react-native-drawer-layout-polyfill": "^1.3.0",
+    "react-native-tab-view": "0.0.58",
+    "react-redux": "^5.0.3",
```
Comment (chirag04) [inline package.json]: this is mostly a non breaking change. major internal refactor.

File: package.json [hunk 3]
```diff
@@ -22,28 +22,27 @@
...
+    "react-static-container": "^1.0.1",
     "react-timer-mixin": "^0.13.3",
-    "redux": "^3.5.2",
-    "redux-batched-actions": "^0.1.2",
-    "reselect": "^2.5.1",
+    "redux": "^3.6.0",
+    "redux-batched-actions": "^0.1.5",
+    "reselect": "^3.0.0",
```
Comment (chirag04) [inline package.json]: I couldn't point out anything major but @skevy @janicduplessis can you guys confirm?

File: package.json [hunk 4]
```diff
@@ -22,28 +22,27 @@
...
     "uuid-js": "^0.7.5",
-    "warning": "^2.1.0"
+    "warning": "^3.0.0"
```
Comment (chirag04) [inline package.json]: just a licence change.

**Comment by chirag04** [comment]
- fix backHandler use

**Comment by sibelius** [comment]
any progress on this?

**Comment by chirag04** [comment]
rebased and updated. @sibelius can you double check the change?

@skevy @janicduplessis any interest here?

**Pull request closed without merging**